src/tool/Plugins.cpp

```diff
@@ -0,0 +1,72 @@
+//
+// This is a derivative work. originally part of the LLVM Project.
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (c) 2023 Vinnie Falco (vinnie.falco@gmail.com)
+//
+// Official repository: https://github.com/cppalliance/mrdox
+//
+
+#include <mrdox/Plugin.hpp>
+#include <mrdox/Version.hpp>
+#include <mrdox/Support/Path.hpp>
+
+#include "lib/Support/GeneratorsImpl.hpp"
+
+#include <llvm/Support/DynamicLibrary.h>
+
+namespace clang {
+namespace mrdox {
+
+class PluginEnvironmentImpl final : public PluginEnvironment
+{
+public:
+    void
+    addGenerator(
+        std::unique_ptr<Generator> generator) override
+    {
+        getGeneratorsImpl().insert(std::move(generator));
+    }
+};
+
+Error loadOnePlugin(
+    std::string pth,
```

Comment:
either use one letter or spell the word, but "pth" is not a great abbreviation, no vowel, unpronounceable. `filePath` is what I use to indicate a path that must identify an ordinary file.